nQueens.java

```diff
@@ -0,0 +1,77 @@
+class Solution {
+
+    //Time Complexity : 0(n!) where n is the input
+    //Space Complexity : 0(n) where n is the recursive stack
```

Comment:
Please check the space complexity.

Reply:
It is correct as per me. Can you please let me know the correct solution?
In line 12, there is an array but it holds only 4 elements which is constant space. Remaining I am not using any additional space.
So could you please highlight where m I using additional space?

Reply:
The board is of size `n x n` and you are using a boolean array for the same.

Reply:
I am not using any boolean array. I am making changes to the original array itself that is passed on to me. So will that count into my space?

Reply:
Line 14: `isTrue = new boolean[n][n];`

Reply:
I am really sorry, I was looking at my word search problem rather that nQueens. 
Yeah you are correct. So n will be the no. of elements so the final space complexity will be
0(n) + 0(l) where n is the no. of elements and l is the recursive stack.
Is the above correct?